### PR TITLE
chore(release): re-enable custom release versions

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Unit tests (`npm test`) were run locally and passed
- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Release-related changes

## What is the current behavior?

When attempting to release Stencil with a custom version, the ability to
choose a custom semver-compliant name is broken. The prompt for
specifying a version gets skipped as such:
Screenshot 2022-11-28 at 8 11 51 AM
<Hit Enter>
Screenshot 2022-11-28 at 8 12 07 AM
Note the empty version number to the right of the 🌹 vermoji

GitHub Issue Number: N/A

## What is the new behavior?

this commit updates the prompts for the release scripts to allow for
custom versions and tags to be specified. an issue was occurring where
two inputs with the same name would only allow the first to be run. this
commit updates the second version & tag prompts to have a unique name,
where the two values are coalesced after either question is asked.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

This PR includes a second commit, e9f80f983510ac1aefcb7337e21dbb0326b64845, which comments out all tagging/pushing/etc. I tested this with that commit of this branch. From the pre-release:

Screenshot 2022-11-28 at 8 13 59 AM
Screenshot 2022-11-28 at 8 14 26 AM
Screenshot 2022-11-28 at 8 14 42 AM
Screenshot 2022-11-28 at 8 18 37 AM

Verified the version gets bumped to the changelog correctly (contents will be wonky, and that's OK, since we target `main` for this PR, it's not actually the v3.0.0-alpha.0 branch):
Screenshot 2022-11-28 at 8 19 51 AM

From the release:
Screenshot 2022-11-28 at 8 21 15 AM
Screenshot 2022-11-28 at 8 21 39 AM

### CODE REVIEW

File: scripts/release.ts [hunk 1]
```diff
@@ -84,13 +84,14 @@ async function prepareRelease(opts: BuildOptions, args: ReadonlyArray<string>, r
     },
     {
       type: 'input',
-      name: 'version',
-      message: 'Version',
+      // this name is intentionally different from 'version' above. if they collide, this input prompt will not run.
```
Comment (alicewriteswrongs) [inline scripts/release.ts:87]:
```suggestion
      // this name is intentionally different from 'version' above to make the `when` check below work properly
      // (this prompt should only run if `version` was not already input)
```

is this an accurate re-phrasing of this comment here? it looks like the `when` check doesn't check that the values are the same, but rather that `version` was not set in the above prompt

**Review by alicewriteswrongs — APPROVED** [review]
just had a question about the comment here, I tried out the functionality locally and it looks good

**Review by alicewriteswrongs — APPROVED** [review]

File: scripts/release.ts [hunk 2]
```diff
@@ -108,7 +111,7 @@ async function prepareRelease(opts: BuildOptions, args: ReadonlyArray<string>, r
     .prompt(prompts)
     .then((answers) => {
       if (answers.confirm) {
-        opts.version = answers.version;
+        opts.version = answers.version || answers.specifiedVersion;
```
Comment (tanner-reits) [inline scripts/release.ts:114]:
```suggestion
        opts.version = answers.version ?? answers.specifiedVersion;
```

**Review by tanner-reits — CHANGES_REQUESTED** [review]
One small change I think we should make for consistency

**Review by tanner-reits — APPROVED** [review]

**Pull request merged**